### PR TITLE
Support converting Python number to IValue in pybind_utils.h

### PR DESCRIPTION
**Pull request by mrshenli**
I ran into the following error when trying to pass a Python int as an arg to `torch::jit::createStackForSchema`, and I think it is due to the missing support for `NumberType` in [toIValue](https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/pybind_utils.h#L448).

> RuntimeError: Missing cases in toIValue for type: Scalar! File a bug report. (toIValue at ../torch/csrc/jit/pybind_utils.h:449)

### CODE REVIEW

File: torch/csrc/jit/pybind_utils.h [hunk 1]
```diff
@@ -446,6 +446,11 @@ inline IValue toIValue(
       return userObj;
     }
     case TypeKind::NumberType:
+      if (py::isinstance<py::int_>(obj)) {
```
Comment (driazati) [inline torch/csrc/jit/pybind_utils.h]: What's the context on the schema you're using that needs to use a `Scalar` type?

The changes seem fine otherwise

Comment (mrshenli) [reply torch/csrc/jit/pybind_utils.h]: I first used `torch::jit::getAllOperatorsFor` to get an operator list for `aten::add`, then trying to match one by passing  (`torch.ones(2, 2)`, `2`) as the `py::args` to `torch::jit::createStackForSchema`.

Comment (mrshenli) [reply torch/csrc/jit/pybind_utils.h]: The arg `2` is somehow recognized as a `NumberType` instead of an `IntType`.

**Review by zdevito — APPROVED** [review]

**Comment by mrshenli** [comment]
The failed test seems to be irrelevant to this PR, as it also occurs in other PRs (e.g. #20995)

**Pull request closed without merging**